WorldCat Discovery.js

```diff
@@ -0,0 +1,185 @@
+{
+	"translatorID": "159c65e6-329f-455e-9caa-e0c80bf8435e",
+	"label": "WorldCat Discovery",
+	"creator": "Andrew Shields",
+	"target": "^https?://[^/]+\\.on\\.worldcat\\.org",
+	"minVersion": "3.0.9",
+	"maxVersion": "",
+	"priority": 90,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcs",
+	"lastUpdated": "2015-01-09 21:18:25"
+}
+
+/*
+WorldCat Discovery translator for Zotero.
+Copyright (C) 2015 Andrew Shields
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+var detectWeb, doWeb, pageIsActionRecord, parseOpenURLData;
+
+detectWeb = function(doc, url) {
+  var activeRecord, isActionRecord, numberOfRecords, record, records, _i, _len;
+  isActionRecord = pageIsActionRecord(url);
+  records = doc.getElementsByClassName('record');
+  numberOfRecords = records.length;
+  activeRecord = doc.getElementsByClassName('record active')[0];
+  if (!(isActionRecord || numberOfRecords)) {
+	return false;
+  } else {
+	if (isActionRecord) {
+	  activeRecord = doc.getElementById('main');
+	  return parseOpenURLData(activeRecord).type;
+	} else {
+	  for (_i = 0, _len = records.length; _i < _len; _i++) {
+		record = records[_i];
+		Zotero.monitorDOMChanges(record, {
+		  attributes: true,
+		  attributeFilter: ['class']
+		});
+	  }
+	  if (activeRecord) {
+		return parseOpenURLData(activeRecord).type;
+	  } else {
+		return 'multiple';
+	  }
+	}
+  }
+};
+
+doWeb = function(doc, url) {
+  var activeRecord, isActionRecord, items, numberOfRecords, oclcNum, record, records, saveItems, _i, _len;
+  isActionRecord = pageIsActionRecord(url);
+  records = doc.getElementsByClassName('record');
+  numberOfRecords = records.length;
+  activeRecord = doc.getElementsByClassName('record active')[0];
+  if (!(isActionRecord || numberOfRecords)) {
+	return false;
```

Comment:
you don't need to check this again. `doWeb` is not called unless `detectWeb` passes